src/Features/Core/Portable/InlineHints/InlineHint.cs

```diff
@@ -36,5 +39,8 @@ namespace Microsoft.CodeAnalysis.InlineHints
         /// </summary>
         public Task<ImmutableArray<TaggedText>> GetDescriptionAsync(Document document, CancellationToken cancellationToken)
             => _getDescriptionAsync?.Invoke(document, cancellationToken) ?? SpecializedTasks.EmptyImmutableArray<TaggedText>();
+
+        public string? GetReplacementText()
+            => _getReplacementText?.Invoke() ?? null;
```

Comment:
why do we need an indirection here?  it should be practically free to just store the TextChange struct.

Reply:
At that point, why would you need the TextChange struct as well?

Reply:
to know what text change to make :)

Reply:
isn't it easier to use the `ITextBuffer.Replace` instead of worrying about `TextChange` etc? We already have the `IWpfTextView` stored so calling `Replace` on the underlying buffer is trivial

Reply:
hints are data, and should work *cross* process.  we would not want them making any changes themselves.